accession/accession.py

```diff
@@ -564,18 +571,220 @@ def accession_steps(self):
+        )[0]
+        qc = self.backend.read_json(qc_file)
+        star_qc_metric = qc.get("star_log_qc")  # what the key is in actual qc json file
+        del star_qc_metric["Started job on"]
+        del star_qc_metric["Started mapping on"]
+        del star_qc_metric["Finished on"]
+        for key, value in star_qc_metric.items():
+            star_qc_metric[key] = string_to_number(value)
+        qc_bytes = self.get_bytes_from_dict(qc)
+        attachment = self.make_attachment_object(
+            qc_bytes, "text/plain", qc_file.filename, ".txt"
+        )
+        star_qc_metric["attachment"] = attachment
+        return self.queue_qc(
+            star_qc_metric, encode_bam_file, "star-quality-metric"
+        )  # backend mapping adding hyphens and removing caps
+
+    def format_reads_by_gene_type_qc(
+        self, qc_dict: Dict, properties_to_report: List
+    ) -> Dict:
+        output = {prop: qc_dict[prop] for prop in properties_to_report}
+        return output
+
+    def make_reads_by_gene_type_qc(self, encode_file: Dict[str, Any], gs_file: GSFile):
+        if self.file_has_qc(encode_file, "GeneTypeQuantificationQualityMetric"):
+            return
+        qc_file = self.analysis.search_down(gs_file.task, "rna_qc", "rnaQC")[0]
+        qc = self.backend.read_json(qc_file)
+        try:
+            reads_by_gene_type_qc_metric = qc["gene_type_count"]
+        except KeyError:
+            self.logger.exception("Something is wrong with rna_qc file")
+            raise
+        output_qc = self.format_reads_by_gene_type_qc(
+            reads_by_gene_type_qc_metric, self.GENE_TYPE_PROPERTIES
+        )
+        qc_bytes = self.get_bytes_from_dict(qc)
+        attachment = self.make_attachment_object(
+            qc_bytes, "text/plain", qc_file.filename, ".txt"
+        )
+        output_qc["attachment"] = attachment
+        return self.queue_qc(
+            output_qc, encode_file, "gene-type-quantification-quality-metric"
+        )
+
+    def make_qc_from_well_formed_json(
+        self,
+        encode_file: Dict[str, Any],
+        gs_file: GSFile,
+        qc_schema_name: str,
+        qc_file_task_output_name: str,
+        qc_dictionary_key: str,
+        qc_schema_name_with_hyphens: str,
+    ):
+        if self.file_has_qc(encode_file, qc_schema_name):
+            return
+        qc_file = self.analysis.get_files(
+            filename=gs_file.task.outputs[qc_file_task_output_name]
+        )[0]
+        qc = self.backend.read_json(qc_file)
+        output_qc = qc.get(qc_dictionary_key)
+        return self.queue_qc(output_qc, encode_file, qc_schema_name_with_hyphens)
+
+    def make_flagstat_qc(
+        self,
+        encode_file: Dict[str, Any],
+        gs_file: GSFile,
+        task_output_name: str,
+        qc_dictionary_key: str,
+        convert_to_string: List = [
+            "mapped_pct",
+            "paired_properly_pct",
+            "singletons_pct",
+        ],
```

Comment:
This looks like it will only be str, so we can annotate it as such.
```suggestion
        convert_to_string: List[str,] = [
            "mapped_pct",
            "paired_properly_pct",
            "singletons_pct",
        ],
```